src/udfs/abstract_udfs.py

```diff
@@ -46,3 +47,14 @@ def classify(self, batch: FrameBatch) -> List[Prediction]:
             List[Prediction]: The predictions made by the classifier
         """
         pass
+
+    @abstractmethod
```

Comment:
We will consider renaming the **classify** method and give it a more generic name that suites other UDFs. For now, just override the **classify** method instead of creating a new abstract method named process_frames.
Refer this commit: The **classify** method now returns a list of **BasePrediction**. (Which should be DepthEstimationResult in your case)
https://github.com/georgia-tech-db/Eva/commit/d7f64d52ed7bf20eb251601b70388d81b57aa058